### PR TITLE
Revised standards

### PR DESCRIPTION
**Pull request by pkruithof**
WIP

Provides an easy way to integrate CS in a repository, along with a common standard.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,94 @@
...
+code before committing it. Be careful however since this can modify code 
+without you seeing it.
+
+```bash
+cp vendor/treehouselabs/standards/hooks/pre-commit .git/pre-commit
+chmod +x .git/pre-commit
+```
+ 
+Of course if you prefer a different method, that's cool. Whatever works. Again,
+the important thing is that it's applied consistently.
```
Comment (cleentfaar) [inline README.md]: My personal method is to use a hotkey in phpstorm that executes php-cs-fixer on the file I'm currently editing, see https://github.com/treehouselabs/standards/blob/master/docs/04-using-phpstorm.md#using-php-cs-fixer, perhaps nice to link to?

File: composer.json [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
     {
       "name": "Peter Kruithof",
       "email": "peter@treehouse.nl"
-    },    {
+    },
+    {
       "name": "Jeroen Fiege",
       "email": "jeroen@treehouse.nl"
     }
   ],
```
Comment (cleentfaar) [inline composer.json]: hm might be nice to get rid of the specific authors while you're at it, it's hard to maintain and besides we all work on it now and then (better to use TreeHouse development or something?)

File: composer.json [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
+    "friendsofphp/php-cs-fixer": "^2.7"
   },
   "require-dev": {
-    "phpunit/phpunit": "~4.4"
+    "phpunit/phpunit": "^6.4"
   },
   "autoload": {
     "psr-4": {
-      "TreeHouse\\Standards\\": "src/TreeHouse/Standards/"
+      "TreeHouse\\": "src/TreeHouse/"
```
Comment (cleentfaar) [inline composer.json]: Curious; could this cause a conflict in certain projects where this prefix is used?

File: composer.json [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
     }
   },
   "autoload-dev": {
     "psr-4": {
-      "TreeHouse\\Standards\\Tests\\": "tests/TreeHouse/Standards/Tests/"
+      "Test\\": "tests/"
```
Comment (cleentfaar) [inline composer.json]: same issue as above perhaps? (again, not sure)

File: src/TreeHouse/CS/Config.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            'declare_strict_types' => true,
+            'dir_constant' => true,
+            'list_syntax' => [
+                'syntax' => 'short',
+            ],
+            'phpdoc_add_missing_param_annotation' => true,
+            'phpdoc_inline_tag' => false,
+            'phpdoc_order' => true,
+            'ternary_to_null_coalescing' => true,
+            'yoda_style' => false,
```
Comment (cleentfaar) [inline src/TreeHouse/CS/Config.php]: `no_useless_return` and `ordered_imports` may be nice as well? Even better; remove these rules completely as it suggests there is some shared standard (I thought you wanted to do that later, which is a bit confusing).

File: README.md (hunk 1 above)
Comment (pkruithof) [reply README.md]: Plugins/helpers are described on the php-cs-fixer page, linked to below.

File: composer.json (hunk 2 above)
Comment (pkruithof) [reply composer.json]: True, I'll update it to dev@treehouse 👍

File: composer.json (hunk 3 above)
Comment (pkruithof) [reply composer.json]: I think it could, better to specify it to the `CS` namespace idd.

File: src/TreeHouse/CS/Config.php (hunk 5 above)
Comment (pkruithof) [reply src/TreeHouse/CS/Config.php]: I used a previous php_cs config as starting point, but I forgot `ordered_imports`, added now 👍 

`no_useless_return` wasn't used before. I'd like to use this as a basis/common set of rules. We can debate and vote for certain rules after this is done.

**Pull request merged**